niimpy/preprocess.py

```diff
@@ -64,21 +89,26 @@ def daily_affect_variability(database,subject):
     the standard deviation of each of the 7 questions. 
     
     NOTE: This function aggregates data by day. 
-    
+
     Parameters:
     --------
-    database: Niimpy database
-    user: string
-    
+    questions: DataFrame with subject data (or database for backwards compatibility)
+    subject: string, optional (backwards compatibility only, in the future do filtering before).
+
     Returns:
     --------
     DLA_mean: mean of the daily affect 
     DLA_std: standard deviation of the daily affect
     """
-    assert isinstance(database, niimpy.database.Data1),"database not given in Niimpy database format"
-    assert isinstance(subject, str),"user not given in string format"
 
-    questions = database.raw(table='AwareHyksConverter', user=subject)
+    # Backwards compatibilty if a database was passed
+    if isinstance(questions, niimpy.database.Data1):
+        questions = questions.raw(table='AwareHyksConverter', user=subject)
+    # Maintain backwards compatibility in the case subject was passed and
+    # questions was *not* a dataframe.
+    elif isinstance(subject, string):
+        questions = questions[questions['user'] == subject]
```

Comment:
Hm, we could also take this and split this into another function, so we don't have to copy and paste it so much.  This can be done later, though, when other functions are being made.